testsuite/tests/parallel/catch_break.ml

```diff
@@ -23,43 +44,65 @@ Success.
 
 *)
 
+let delay = 0.001 (* 1 ms *)
+let fuel = Atomic.make 1000 (* 1s *)
+
 let print = if verbose then print_endline else fun _ -> ()
 
+let ready_count = Atomic.make 0
+
+(* Does not poll *)
+
+let sleep () =
+  if Atomic.get fuel <= 0 then (
+    print "[Reached max attempts without succeeding]";
+    Unix._exit 1
+  );
+  Atomic.decr fuel;
+  Unix.sleepf delay
+
+let rec wait n =
+  if Atomic.get ready_count <> n then (
+    sleep ();
+    wait n
+  )
+
 let break_trap s =
   begin
-    try while true do () done
-    with Sys.Break -> print "[Sys.Break caught]";
+    try Atomic.incr ready_count; while true do () done
+    with Sys.Break -> print "[Sys.Break caught]"
   end;
-  print s
+  print s;
+  Atomic.decr ready_count
+
+(* Simulate repeated Ctrl-C *)
+let interruptor_domain () =
+  Domain.spawn @@ fun () ->
+  ignore (Thread.sigmask Unix.SIG_BLOCK [Sys.sigint]);
+  let kill () = sleep () ; Unix.kill (Unix.getpid ()) Sys.sigint in
+  wait 2;
+  kill ();
+  wait 1;
+  kill ();
+  wait 2;
+  kill ()
```

Comment:
It is not obvious to me why we have this `2 1 2` sequence of waited-on `ready_count` values. I suppose this is suppose to mirror the `run ()` code, but the mapping is not obvious. Maybe we could have a comment on each `wait` that details which of the `break_trap ()` from `run ()` we expect to be active at this point?